lib/src/main/java/io/ably/lib/realtime/AblyRealtime.java

```diff
@@ -72,10 +72,19 @@ public void close() {
 	public class Channels extends HashMap<String, Channel> {
 		public Channels() {
 			/* remove all channels when the connection is closed, to avoid stalled state */
-			connection.on(ConnectionState.closed, new ConnectionStateListener() {
+			connection.on(new ConnectionStateListener() {
```

Comment:
Let's say you do this:

```
AblyRealtime realtime = new AblyRealtime(...);
realtime.connection.on(someListener);
realtime.connection.on(someOtherListener);

// Now I disable all my listeners, for example because my app goes to the background or whatever.
realtime.connection.off();
```

As a side effect, I've also disabled the listener registered [here](https://github.com/ably/ably-java/pull/135/files/c5a158e96d7dd0d73e35cbdf18d920ef30d2cd8b#diff-b8b1ac0a34230df724c2576f32005521R75), so if the connection state transitions to FAILED, channels won't transition to FAILED anymore.

What I've done is to have a public EventEmitter, the one that gets called when you do `realtime.connection.on(...)`, and a private/internal EventEmitter for internal logic like this, so that operations on the public one don't affect internal functionality.